src/Models.js

```diff
@@ -30,7 +30,7 @@
     calcExpTillNextLevel() {
       const levelExp = this.exp - expTable[this.level];
       const totalExpNeeded = expTable[this.level + 1] - expTable[this.level];
-      this.expTillNextLevel = totalExpNeeded - levelExp;
```

Comment:
expTillNextLevel name is in this case confusing. You should add two class fields instead one with wrong name.